bson/bsonrw/extjson_wrappers.go

```diff
@@ -199,6 +203,9 @@ func (ejv *extJSONValue) parseDateTime() (int64, error) {
 
 func parseDatetimeString(data string) (int64, error) {
 	t, err := time.Parse(rfc3339Milli, data)
```

Comment:
Rather than attempt to parse each of these by name individually, why not have a top-level slice of all the formats and just iterate over them from front to back. Adding new formats in the future would be as simple as adding it into the slice... no other code changes necessary.

In addition, this would make it possible to add a new format through a global `func RegisterDateTimeFormat` if you wanted to make this flexible.